src/cy_serial_bridge/cy_scb_context.py

```diff
@@ -110,79 +167,49 @@ def list_devices(
-                # USB CDC mode
-                usb_cdc_interface_settings = cfg[0][0]
-                cdc_data_interface_settings = cfg[1][0]
-                mfg_interface_settings = cfg[2][0]
-
-                # Check USB CDC interface
-                if usb_cdc_interface_settings.getSubClass() != 0x2:
-                    continue
-
-                # Check CDC Data interface
-                if cdc_data_interface_settings.getClass() != 0x0A or cdc_data_interface_settings.getSubClass() != 0x0:
-                    continue
-
-                curr_cytype = CyType.UART_CDC
-
-            else:
-                # USB vendor mode
-                scb_interface_settings = cfg[0][0]
-                mfg_interface_settings = cfg[1][0]
-
-                # Check SCB interface -- the Class should be 0xFF (vendor defined/no rules)
-                # and the SubClass value gives the CyType
-                if scb_interface_settings.getClass() != USBClass.VENDOR:
-                    continue
-                if scb_interface_settings.getSubClass() not in {
-                    CyType.UART_VENDOR.value,
-                    CyType.SPI.value,
-                    CyType.I2C.value,
-                }:
-                    continue
-
-                # Check SCB endpoints
-                if scb_interface_settings.getNumEndpoints() != 3:
-                    continue
-                # Bulk host-to-dev endpoint
-                if (
-                    scb_interface_settings[0].getAddress() != 0x01
-                    or (scb_interface_settings[0].getAttributes() & 0x3) != 2
-                ):
-                    continue
-                # Bulk dev-to-host endpoint
-                if (
-                    scb_interface_settings[1].getAddress() != 0x82
-                    or (scb_interface_settings[1].getAttributes() & 0x3) != 2
-                ):
-                    continue
-                # Interrupt dev-to-host endpoint
-                if (
-                    scb_interface_settings[2].getAddress() != 0x83
-                    or (scb_interface_settings[2].getAttributes() & 0x3) != 3
-                ):
-                    continue
-
-                curr_cytype = CyType(scb_interface_settings.getSubClass())
-
-            # Check manufacturer interface.
-            # It has a defined class/subclass and has no endpoints
-            if mfg_interface_settings.getClass() != 0xFF:
-                continue
-            if mfg_interface_settings.getSubClass() != CyType.MFG:
-                continue
-            if mfg_interface_settings.getNumEndpoints() != 0:
+            for i in range(cfg.getNumInterfaces()):
+                type = self.identify_interface(cfg[i])
+                if type == None:
+                    pass # TODO verbose output
+                else:
+                    match(type):
+                        case CyType.UART_CDC: # TODO we could have two of these!
```

Comment:
I would like to handle this case a little bit better in this function.  I think what would make sense would be to add two DiscoveredDevice entries to the output when we see a dual channel device.  Either that, or make DiscoveredDevice able to hold two different channels from one device.